src/sql-parser/src/parser.rs

```diff
@@ -7678,6 +7698,51 @@ impl<'a> Parser<'a> {
         )
     }
 
+    /// Bail out if the current token is not a privilege object type in the plural form, or consume and
+    /// return it if it is.
+    fn expect_plural_system_object_type_for_privileges(
+        &mut self,
+    ) -> Result<SystemObjectType, ParserError> {
+        if let Some(object_type) = self.parse_one_of_keywords(&[VIEWS, SOURCES]) {
+            return parser_err!(
+                self,
+                self.peek_prev_pos(),
+                format!("For object type {object_type}, you must specify 'TABLES'")
+            );
+        }
+        if self.parse_keywords(&[MATERIALIZED, VIEWS]) {
+            self.prev_token();
+            return parser_err!(
+                self,
+                self.peek_prev_pos(),
+                format!("For object type MATERIALIZED VIEWS, you must specify 'TABLES'")
+            );
+        }
```

Comment:
Why do we special case Views, Sources, and Materialized Views like this?

Reply:
It's just a weird quirk of RBAC that all relations are considered tables. I'm not sure if it comes from PostgreSQL or the SQL Standard originally.

```
postgres=# CREATE VIEW v AS SELECT 1;
CREATE VIEW
postgres=# GRANT SELECT ON VIEW v TO joe;
ERROR:  syntax error at or near "v"
LINE 1: GRANT SELECT ON VIEW v TO joe;
                             ^
postgres=# GRANT SELECT ON TABLE v TO joe;
GRANT

```